### PR TITLE
[#9255] Migrate Instructor Course Details Page to Angular

### PR DESCRIPTION
**Pull request by jacoblipech**
Part of #9255

Migrated part of `InstructorCourseDetailsPageAction` to `GetInstructorCourseDetailsAction`. The new endpoint is at `GET` `/courses/details` and requires 1 `GET` parameters, courseid. 

Migrated `InstructorCourseStudentDeleteAllAction` to `DeleteInstructorCourseAllStudentsAction`. The endpoint `DELETE` `/courses/details/deleteAllStudents` with 1 compulsory GET param, courseid.

Migrated `InstructorCourseStudentDeleteAllAction` to `RemindInstructorCourseStudentsAction`. The endpoint `POST` `/courses/details/remindAllStudents` with 1 compulsory GET param, courseid.

Migrated `InstructorCourseStudentListDownloadAction` to `DownloadInstructorCourseStudentsAction`. The endpoint `GET` `/courses/details/downloadAllStudents` with 1 compulsory GET param, courseid.

Populated `StudentList` component completed in PR #9317 

Current UI of the page:

screenshot 2019-01-16 at 3 26 30 pm

screenshot 2019-01-22 at 11 47 14 am

screenshot 2019-01-22 at 11 57 20 am

screenshot 2019-01-20 at 2 43 39 am

### CODE REVIEW

**Comment by jacoblipech** [comment]
@wkurniawan07 Updated the comments. Can help me to do a first round of review? Thanks.

File: src/main/java/teammates/ui/newcontroller/GetInstructorCourseDetailsAction.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    /**
+     * Output format for {@link GetInstructorCourseDetailsAction}.
+     */
+    public static class CourseInfo extends ActionResult.ActionOutput {
+        private final CourseDetailsBundle courseDetails;
+        private final InstructorAttributes currentInstructor;
+        private final List<InstructorAttributes> instructors;
+        private final List<StudentListSectionData> sections;
+        private final boolean hasSection;
+        private String studentListHtmlTableAsString;
```
[the remark this answers is not in the document]
Comment (jacoblipech) [reply src/main/java/teammates/ui/newcontroller/GetInstructorCourseDetailsAction.java]: Do you suggest that I should populate the data and convert to HTML in the frontend then?

**Comment by jacoblipech** [comment]
@wkurniawan07 Made the necessary changes. Ready for review.

**Comment by jacoblipech** [comment]
@wkurniawan07 Ready for review.

File: src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.html [hunk 2]
```diff
@@ -1,3 +1,143 @@
...
+    </div>
+    <div class="modal-footer">
+      <button type="button" class="modal-btn-cancel btn-lg btn-default" (click)="modal.dismiss()">Cancel</button>
+      <button type="button" class="modal-btn-ok btn-lg btn-danger" (click)="deleteAllStudentsFromCourse(courseDetails.course.id); modal.dismiss()">OK</button>
+    </div>
+  </ng-template>
+  <ng-template #studentTableModal let-modal>
+    <div class="modal-header">
+      <span>
+        Tips: <span class="btn-link" (click)="copyContent(courseStudentListAsCsv)">Click here</span>
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.html]: The problem with this is you're copying the CSV content as is, not actually copying the whole table (expected: tab as separator, actual: comma as separator). This needs to be fixed after this.

File: src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.html [hunk 3]
```diff
@@ -1,3 +1,143 @@
...
+        to copy the table, then <kbd>Ctrl + V</kbd> or <kbd>⌘ + V</kbd> to PASTE to your Excel Workbook.
+      </span>
+      <button type="button" class="close" aria-label="Close" (click)="modal.dismiss('Cross click')">
+        <span aria-hidden="true">&times;</span>
+      </button>
+    </div>
+    <div class="modal-body">
+      <tm-ajax-preload *ngIf="loading"></tm-ajax-preload>
+      <div [innerHTML]="convertToHtmlTable(courseStudentListAsCsv)"></div>
+      <b>{{ isAjaxSuccess ? '' : 'Failed to load. Please reopen modal.' }}</b>
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.html]: Closing-then-reopening the modal when something fails does not sound very user friendly. As a user I would expect something inside the modal itself to be clickable to reload.

**Pull request merged**